src/hooks/use-aggrid-sort.ts

```diff
@@ -32,22 +68,61 @@ export const getSortValue = (sortWay: number) => {
 };
 
 export const useAgGridSort = (
-    initSortConfig: SortConfigType
+    initSortConfig: ColumnSortConfig
 ): SortPropsType => {
-    const { colKey: initColKey, sortWay: initSortWay } = initSortConfig;
+    const { sortWay: initSortWay } = initSortConfig;
 
-    const [sortConfig, setSortConfig] = useState<SortConfigType>({
-        colKey: initColKey,
-        sortWay: initSortWay,
-    });
+    const [sortConfig, setSortConfig] = useState<SortConfigType>([
+        initSortConfig,
+    ]);
 
     const onSortChanged = useCallback(
-        (colKey: string, sortWay: number) => setSortConfig({ colKey, sortWay }),
-        []
+        (colKey: string, sortWay: number, secondary: boolean) => {
+            const primarySort = getPrimarySort(sortConfig);
+            const secondarySort = getSecondarySort(sortConfig);
+
+            if (secondary) {
+                setSortConfig([
+                    primarySort,
+                    {
+                        colKey: colKey,
+                        sortWay: sortWay,
+                        secondary: true,
+                    },
+                ]);
+            } else {
+                if (secondarySort) {
+                    setSortConfig([
+                        {
+                            colKey: colKey,
+                            sortWay: sortWay,
+                            secondary: false,
+                        },
+                        secondarySort,
+                    ]);
+                } else {
+                    setSortConfig([
+                        {
+                            colKey: colKey,
+                            sortWay: sortWay,
+                            secondary: false,
+                        },
+                    ]);
+                }
+            }
+        },
+        [sortConfig]
```

Comment:
It's not really what I was thinking about.. what about that instead ? I am right ?
```suggestion
        (colKey: string, sortWay: number, secondary: boolean) => {
            setSortConfig(
                sortConfig
                    .filter((sort) => sort.secondary !== secondary)
                    .concat({ colKey, sortWay, secondary })
            );
        },
        [sortConfig]
```